pkg/kubelet/kuberuntime/kuberuntime_gc.go

```diff
@@ -338,9 +338,26 @@ func (cgc *containerGC) evictPodLogsDirectories(allSourcesReady bool) error {
 	logSymlinks, _ := osInterface.Glob(filepath.Join(legacyContainerLogsDir, fmt.Sprintf("*.%s", legacyLogSuffix)))
 	for _, logSymlink := range logSymlinks {
 		if _, err := osInterface.Stat(logSymlink); os.IsNotExist(err) {
-			err := osInterface.Remove(logSymlink)
+			firstlinkdstPath, err := os.Readlink(logSymlink)
 			if err != nil {
-				glog.Errorf("Failed to remove container log dead symlink %q: %v", logSymlink, err)
+				glog.Errorf("Failed to resolve the log symlink %q: %v", logSymlink, err)
+				return nil
```

Comment:
if os.Readlink throws an error, shouldn't we try to remove logSymlink before return ?

Reply:
@bruceauyeung I think if os.Readlink throwing an error means the logSymlink has already been removed, so there is no need to remove it again. Correct me if I'm wrong.